diff --git a/src/com/brockoli/android/hsdroid/MainActivity.java b/src/com/brockoli/android/hsdroid/MainActivity.java
index 4ebf2cb..da9fdcf 100644
--- a/src/com/brockoli/android/hsdroid/MainActivity.java
+++ b/src/com/brockoli/android/hsdroid/MainActivity.java
@@ -199,7 +199,6 @@ LoaderCallbacks<Cursor>{
 		SlidingMenu menu = getSlidingMenu();
 		menu.setBehindOffsetRes(R.dimen.actionbar_home_width);
 		menu.setBehindScrollScale(0.25f);
-		menu.setShadowDrawable(R.drawable.defaultshadow);
 		menu.setShadowWidth(12);	
 	}
 
