diff --git a/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java b/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java
index d94d995..ca46e26 100644
--- a/src/com/seb/SLWP/SLWP.java	
+++ b/src/com/seb/SLWP/SLWP.java	
@@ -590,7 +590,7 @@ public class SLWP extends GLWallpaperService implements
 		return mGle;
 	}
 
-	class GlEngine extends GLEngine {
+	class GlEngine extends GLWallpaperService.GLEngine {
 
 		private static final float TOUCH_SCALE_FACTOR = 1E-3f;
 		private float mPreviousX = 0.0f;
