commit 3e0b60afc74bed5d0d864ae08a63d5727b1c1b8e
Author: David Black <david8black@gmail.com>
Date:   Mon Mar 18 18:43:31 2013 +0000

    patch

diff --git a/ earth-live-wallpaper/SLWP/res/values-es/strings.xml b/ earth-live-wallpaper/SLWP/res/values-es/strings.xml
index a3f9b91..5c89079 100644
--- a/ earth-live-wallpaper/SLWP/res/values-es/strings.xml	
+++ b/ earth-live-wallpaper/SLWP/res/values-es/strings.xml	
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
 	<string name="hello">Hello World, SLWP!</string>
-	<string name="app_name">EarthRot</string>
 	<string name="description">rotación Globe (OpenGL VBO)</string>
 	<string name="p_speed">velocidad</string>
 	<string name="p_speedsum">Elija la velocidad de rotación de la Tierra
diff --git a/ earth-live-wallpaper/SLWP/res/values/strings.xml b/ earth-live-wallpaper/SLWP/res/values/strings.xml
index 501db7e..99d20aa 100644
--- a/ earth-live-wallpaper/SLWP/res/values/strings.xml	
+++ b/ earth-live-wallpaper/SLWP/res/values/strings.xml	
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
 	<string name="hello">Hello World, SLWP!</string>
-	<string name="app_name">EarthRot</string>
+	<string name="app_name">Earth Live Wallpaper</string>
 	<string name="description">Rotating earth globe (Opengl VBO)</string>
 
 
diff --git a/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/Prefs.java b/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/Prefs.java
index 900ff3d..cb0a5c7 100755
--- a/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/Prefs.java	
+++ b/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/Prefs.java	
@@ -75,11 +75,18 @@ public class Prefs extends PreferenceActivity implements
 		bg.setOnPreferenceChangeListener(this);
 		initMapList();
 		Button bt = (Button) findViewById(R.id.bt_donate);
+		final PreferenceActivity myact = this;
 		bt.setOnClickListener(new OnClickListener() {
 			@Override
 			public void onClick(View v) {
-				startActivity(new Intent(Intent.ACTION_VIEW, Uri
-						.parse("market://search?q=pub:unixseb")));
+				try {
+					startActivity(new Intent(Intent.ACTION_VIEW, Uri
+ 						.parse("market://search?q=pub:unixseb")));
+				} catch (Exception e) {
+					Log.e("SLWP", "Cannot open 'market:' link", e);
+					Toast.makeText(myact, "Cannot launch Market", Toast.LENGTH_SHORT).show();
+				}
+
 			}
 		});
 		Button bth = (Button) findViewById(R.id.bt_help);
diff --git a/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java b/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java
index d94d995..d697662 100644
--- a/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java	
+++ b/ earth-live-wallpaper/SLWP/src/com/seb/SLWP/SLWP.java	
@@ -590,7 +590,7 @@ public class SLWP extends GLWallpaperService implements
 		return mGle;
 	}
 
-	class GlEngine extends GLEngine {
+	class GlEngine extends GLWallpaperService.GLEngine {
 
 		private static final float TOUCH_SCALE_FACTOR = 1E-3f;
 		private float mPreviousX = 0.0f;
@@ -841,4 +841,4 @@ public class SLWP extends GLWallpaperService implements
 		}
 	}
 
-}
\ No newline at end of file
+}
