diff --git a/scripts/android-build-freerdp.sh b/scripts/android-build-freerdp.sh
index b27fa6c9b..7c504b11e 100755
--- a/scripts/android-build-freerdp.sh
+++ b/scripts/android-build-freerdp.sh
@@ -111,7 +111,7 @@ do
         then
             common_run bash $SCRIPT_PATH/android-build-openh264.sh \
                 --src $BUILD_SRC/openh264 --dst $BUILD_DST \
-                --ndk $ANDROID_NDK \
+                --ndk $ANDROID_NDK_OPENH264 \
                 --arch $ARCH \
                 --target $NDK_TARGET \
                 --tag $OPENH264_TAG
