diff --git a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
index 26a3016ec4..30f122d093 100644
--- a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
+++ b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
@@ -253,6 +253,7 @@ public class LibFreeRDP
 
 		args.add("/v:" + hostname);
 		args.add("/port:" + port);
+		args.add("/tls:seclevel:0");
 
 		arg = bookmark.getUsername();
 		if (!arg.isEmpty())
