--- a/TumCampusApp/src/de/tum/in/tumcampusapp/activities/StartupActivity.java	2013-11-23 12:45:50.093747730 +0100
+++ b/TumCampusApp/src/de/tum/in/tumcampusapp/activities/StartupActivity.java	2013-11-23 16:31:20.752918783 +0100
@@ -6,7 +6,7 @@
 import android.os.StrictMode;
 import android.util.Log;
 
-import com.bugsense.trace.BugSenseHandler;
+//import com.bugsense.trace.BugSenseHandler;
 
 import de.tum.in.tumcampusapp.auxiliary.DemoModeStartActivity;
 
@@ -18,18 +18,18 @@
  */
 public class StartupActivity extends Activity {
 	public static final boolean DEMO_MODE = false;
-	public static final boolean TRACK_ERRORS_WITH_BUG_SENSE = true;
+	public static final boolean TRACK_ERRORS_WITH_BUG_SENSE = false;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
-		// Init a Bug Report to https://www.bugsense.com
+		/** Init a Bug Report to https://www.bugsense.com
 		if (TRACK_ERRORS_WITH_BUG_SENSE) {
 			Log.d(this.getClass().getSimpleName(), "BugSense initialized");
 			BugSenseHandler.initAndStartSession(this, "19d18764");
 		}
-
+**/
 		// Workaround for new API version. There was a security update which
 		// disallows applications to execute HTTP request in the GUI main
 		// thread.
