diff -r 4af3b30af2a6 app/src/main/java/net/kervala/comicsreader/Album.java
--- a/app/src/main/java/net/kervala/comicsreader/Album.java	Fri Dec 27 16:48:15 2013 +0100
+++ b/app/src/main/java/net/kervala/comicsreader/Album.java	Wed Jan 01 19:43:24 2014 +0100
@@ -56,7 +56,6 @@
 
 	static final int ALBUM_TYPE_NONE = 0;
 	static final int ALBUM_TYPE_CBZ = 1;
-	static final int ALBUM_TYPE_CBR = 2;
 	static final int ALBUM_TYPE_CBT = 3;
 	static final int ALBUM_TYPE_FOLDER = 4;
 
@@ -76,10 +75,6 @@
 			return new CbzAlbum();
 		}
 
-		if (RarFile.isLoaded() && CbrAlbum.isValid(filename)) {
-			return new CbrAlbum();
-		}
-
 		if (CbtAlbum.isValid(filename)) {
 			return new CbtAlbum();
 		}
@@ -113,11 +108,6 @@
 			return ALBUM_TYPE_CBZ;
 		}
 
-		// file is a cbr
-		if (RarFile.isLoaded() && CbrAlbum.isValid(filename)) {
-			return ALBUM_TYPE_CBR;
-		}
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) {
 			return ALBUM_TYPE_CBT;
@@ -138,7 +128,6 @@
 		switch(getType(uriString))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getMimeType(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getMimeType(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getMimeType(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getMimeType(filename);
 			default: break;
@@ -161,9 +150,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (RarFile.isLoaded() && CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -187,9 +173,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (RarFile.isLoaded() && CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -203,9 +186,6 @@
 		// file is a cbz
 		if (CbzAlbum.askConfirm(filename)) return true;
 
-		// file is a cbr
-		if (RarFile.isLoaded() && CbrAlbum.askConfirm(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.askConfirm(filename)) return true;
 
@@ -219,7 +199,6 @@
 		switch(getType(filename))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getTitle(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getTitle(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getTitle(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getTitle(filename);
 			default: break;

