diff -r 6650f062edf6 src/net/kervala/comicsreader/Album.java
--- a/src/net/kervala/comicsreader/Album.java	Mon Jun 10 20:46:47 2013 +0200
+++ b/src/net/kervala/comicsreader/Album.java	Tue Jul 30 12:26:15 2013 +0100
@@ -57,7 +57,6 @@
 
 	static final int ALBUM_TYPE_NONE = 0;
 	static final int ALBUM_TYPE_CBZ = 1;
-	static final int ALBUM_TYPE_CBR = 2;
 	static final int ALBUM_TYPE_CBT = 3;
 	static final int ALBUM_TYPE_FOLDER = 4;
 
@@ -66,10 +65,6 @@
 			return new CbzAlbum();
 		}
 
-		if (CbrAlbum.isValid(filename)) {
-			return new CbrAlbum();
-		}
-
 		if (CbtAlbum.isValid(filename)) {
 			return new CbtAlbum();
 		}
@@ -103,11 +98,6 @@
 			return ALBUM_TYPE_CBZ;
 		}
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) {
-			return ALBUM_TYPE_CBR;
-		}
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) {
 			return ALBUM_TYPE_CBT;
@@ -128,7 +118,6 @@
 		switch(getType(uriString))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getMimeType(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getMimeType(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getMimeType(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getMimeType(filename);
 			default: break;
@@ -151,9 +140,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -177,9 +163,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -193,9 +176,6 @@
 		// file is a cbz
 		if (CbzAlbum.askConfirm(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.askConfirm(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.askConfirm(filename)) return true;
 
@@ -209,7 +189,6 @@
 		switch(getType(filename))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getTitle(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getTitle(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getTitle(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getTitle(filename);
 			default: break;
diff -r 6650f062edf6 src/net/kervala/comicsreader/FolderAlbum.java
--- a/src/net/kervala/comicsreader/FolderAlbum.java	Mon Jun 10 20:46:47 2013 +0200
+++ b/src/net/kervala/comicsreader/FolderAlbum.java	Tue Jul 30 12:26:15 2013 +0100
@@ -58,7 +58,7 @@
 			// build a full absolute path
 			filename = f.getAbsolutePath();  
 
-			return CbzAlbum.isValid(filename) || CbrAlbum.isValid(filename) || CbtAlbum.isValid(filename) || f.isDirectory();
+			return CbzAlbum.isValid(filename) || CbtAlbum.isValid(filename) || f.isDirectory();
 		}
 	}
 	
