diff -r bd66e1a493b7 -r e16370356b7a src/net/kervala/comicsreader/Album.java
--- a/src/net/kervala/comicsreader/Album.java	Fri Jan 11 11:23:07 2013 +0100
+++ b/src/net/kervala/comicsreader/Album.java	Wed Feb 06 14:44:25 2013 +0000
@@ -65,10 +65,6 @@
 			return new CbzAlbum();
 		}
 
-		if (CbrAlbum.isValid(filename)) {
-			return new CbrAlbum();
-		}
-
 		if (CbtAlbum.isValid(filename)) {
 			return new CbtAlbum();
 		}
@@ -102,11 +98,6 @@
 			return ALBUM_TYPE_CBZ;
 		}
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) {
-			return ALBUM_TYPE_CBR;
-		}
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) {
 			return ALBUM_TYPE_CBT;
@@ -127,7 +118,6 @@
 		switch(getType(uriString))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getMimeType(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getMimeType(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getMimeType(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getMimeType(filename);
 			default: break;
@@ -150,9 +140,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -176,9 +163,6 @@
 		// file is a cbz
 		if (CbzAlbum.isValid(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.isValid(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.isValid(filename)) return true;
 
@@ -192,9 +176,6 @@
 		// file is a cbz
 		if (CbzAlbum.askConfirm(filename)) return true;
 
-		// file is a cbr
-		if (CbrAlbum.askConfirm(filename)) return true;
-
 		// file is a cbt
 		if (CbtAlbum.askConfirm(filename)) return true;
 
@@ -208,7 +189,6 @@
 		switch(getType(filename))
 		{
 			case ALBUM_TYPE_CBZ: return CbzAlbum.getTitle(filename);
-			case ALBUM_TYPE_CBR: return CbrAlbum.getTitle(filename);
 			case ALBUM_TYPE_CBT: return CbtAlbum.getTitle(filename);
 			case ALBUM_TYPE_FOLDER: return FolderAlbum.getTitle(filename);
 			default: break;
diff -r bd66e1a493b7 -r e16370356b7a src/net/kervala/comicsreader/FolderAlbum.java
--- a/src/net/kervala/comicsreader/FolderAlbum.java	Fri Jan 11 11:23:07 2013 +0100
+++ b/src/net/kervala/comicsreader/FolderAlbum.java	Wed Feb 06 14:44:25 2013 +0000
@@ -52,7 +52,7 @@
 	public static class OtherFilter implements FilenameFilter {
 		public boolean accept(File dir, String filename) {
 			File f = new File(dir, filename);
-			return CbzAlbum.isValid(filename) || CbrAlbum.isValid(filename) || f.isDirectory();
+			return CbzAlbum.isValid(filename) || f.isDirectory();
 		}
 	}
 	
