diff --git a/build.gradle b/build.gradle
index feb52d7..691c610 100644
--- a/build.gradle
+++ b/build.gradle
@@ -8,10 +8,20 @@ buildscript {
 }
 apply plugin: 'android'
 
+repositories {
+    mavenCentral()
+}
+
 dependencies {
     compile fileTree(dir: 'libs', include: '*.jar')
     compile 'com.android.support:support-v4:19.1.+'
     compile 'com.android.support:appcompat-v7:19.1.+'
+    compile 'ch.acra:acra:4.4.0'
+    compile 'org.shredzone.flattr4j:flattr4j-core:2.7'
+    compile 'com.google.code.gson:gson:2.2.4'
+    compile 'com.google.guava:guava:16.0'
+    compile 'com.google.code.findbugs:jsr305:2.0.3'
+    compile 'com.mcxiaoke.volley:library:1.0.6'
 }
 
 Properties props = new Properties()
