From bbb04d0b0c5a698af272e4c002641ac22f0c2bc4 Mon Sep 17 00:00:00 2001
From: Marcus Hoffmann <m.hoffmann@cartelsol.com>
Date: Mon, 28 Aug 2017 23:25:18 +0200
Subject: [PATCH 2/2] add clientid and secret

---
 app/build.gradle | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/build.gradle b/app/build.gradle
index 5947b592..c0845589 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -32,8 +32,8 @@ android {
         versionCode 54
         versionName "4.2.0"
 
-        buildConfigField 'String', 'CLIENT_ID', ClientId
-        buildConfigField 'String', 'CLIENT_SECRET', ClientSecret
+        buildConfigField 'String', 'CLIENT_ID', '"f0f32f2ee4a709ba3640"'
+        buildConfigField 'String', 'CLIENT_SECRET', '"f94cffda88486696c2762ad986bc0b8b830c9cff"'
     }
 
     buildTypes {
-- 
2.11.0

