diff --git a/build.gradle b/build.gradle
index 5150e12..6cd42b1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,6 @@ buildscript {
 
 repositories {
     maven { url 'http://dl.bintray.com/populov/maven' }
-    maven { url "http://www.bugsense.com/gradle/" }
     maven { url 'https://github.com/kageiit/maven-repo/raw/master' }
     mavenCentral()
 }
@@ -50,7 +49,6 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: '*.jar')
 
-    compile "com.bugsense.trace:bugsense:3.6"
     compile 'com.android.support:support-v4:19.+'
     compile 'com.actionbarsherlock:actionbarsherlock:4.4.0@aar'
     compile 'com.viewpagerindicator:library:2.4.1@aar'
diff --git a/src/com/gh4a/Gh4Application.java b/src/com/gh4a/Gh4Application.java
index 0e894d6..f56a934 100644
--- a/src/com/gh4a/Gh4Application.java
+++ b/src/com/gh4a/Gh4Application.java
@@ -47,7 +47,6 @@ import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
 import android.content.res.Configuration;
 import android.graphics.Typeface;
 
-import com.bugsense.trace.BugSenseHandler;
 import com.gh4a.activities.Github4AndroidActivity;
 
 /**
@@ -99,7 +98,6 @@ public class Gh4Application extends Application implements OnSharedPreferenceCha
         selectTheme(prefs.getInt("theme", Constants.Theme.DARK));
         prefs.registerOnSharedPreferenceChangeListener(this);
 
-        BugSenseHandler.initAndStartSession(this, "1e6a83ae");
 
         mPt = new PrettyTime();
 
