diff --git a/build.gradle b/build.gradle
index 7e2eff9..75b7624 100644
--- a/build.gradle
+++ b/build.gradle
@@ -4,7 +4,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:0.12.+'
+        classpath 'com.android.tools.build:gradle:1.0.0'
     }
 }
 
@@ -37,16 +26,6 @@ android {
         versionName "3.7.1"
     }
 
-    buildTypes {
-        debug {
-            zipAlign true
-        }
-        release {
-            zipAlign true
-            signingConfig signingConfigs.playStore
-        }
-    }
-
     sourceSets {
         main {
             manifest.srcFile 'AndroidManifest.xml'
