diff --git a/build.gradle b/build.gradle
index 7e2eff9..39eed52 100644
--- a/build.gradle
+++ b/build.gradle
@@ -10,9 +10,6 @@ buildscript {
 
 repositories {
     maven { url 'http://dl.bintray.com/populov/maven' }
-    /* The MINT jar should be present at this repo according
-       to the docs, but it gives a 404 error
-    maven { url "http://mint.splunk.com/gradle/" } */
     maven { url 'https://github.com/kageiit/maven-repo/raw/master' }
     mavenCentral()
 }
@@ -60,8 +57,6 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: '*.jar')
 
-    /* see above
-    compile "com.splunk.mint:mint:4.0" */
     compile 'com.android.support:support-v4:19.+'
     compile 'com.actionbarsherlock:actionbarsherlock:4.4.0@aar'
     compile 'com.viewpagerindicator:library:2.4.1@aar'
diff --git a/src/com/gh4a/Gh4Application.java b/src/com/gh4a/Gh4Application.java
index 245445e..e2da520 100644
--- a/src/com/gh4a/Gh4Application.java
+++ b/src/com/gh4a/Gh4Application.java
@@ -46,7 +46,6 @@ import android.content.res.Configuration;
 import android.graphics.Typeface;
 
 import com.gh4a.activities.SettingsActivity;
-import com.splunk.mint.Mint;
 
 /**
  * The Class Gh4Application.
@@ -97,9 +96,6 @@ public class Gh4Application extends Application implements OnSharedPreferenceCha
         selectTheme(prefs.getInt(SettingsActivity.KEY_THEME, Constants.Theme.DARK));
         prefs.registerOnSharedPreferenceChangeListener(this);
 
-        Mint.disableNetworkMonitoring();
-        Mint.initAndStartSession(this, "1e6a83ae");
-
         mPt = new PrettyTime();
 
         mClient = new DefaultClient();
