diff --git a/shared.gradle b/shared.gradle
index dff2729..984fb05 100644
--- a/shared.gradle
+++ b/shared.gradle
@@ -8,8 +8,7 @@ android {
 }

 dependencies {
-    compile 'com.google.guava:guava:22.0'
-    compile 'com.google.android.gms:play-services:8.4.0'
+    compile 'com.google.guava:guava:22.0-android'
     compile 'com.android.support:support-v4:27.0.0'
     compile 'com.android.support:appcompat-v7:27.0.0'
 }
diff --git a/talkback/src/main/java/overlay/base/com/android/talkback/TalkBackPreferencesActivity.java b/talkback/src/main/java/overlay/base/com/android/talkback/TalkBackPreferencesActivity.java
index d630c86..ad5d660 100644
--- a/talkback/src/main/java/overlay/base/com/android/talkback/TalkBackPreferencesActivity.java
+++ b/talkback/src/main/java/overlay/base/com/android/talkback/TalkBackPreferencesActivity.java
@@ -70,8 +70,6 @@ import com.google.android.accessibility.utils.ServiceStateListener;
 import com.google.android.accessibility.utils.SharedPreferencesUtils;
 import com.google.android.accessibility.utils.WebActivity;
 import com.google.android.clockwork.remoteintent.RemoteIntent;
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GooglePlayServicesUtil;
 import java.util.List;

 /** Activity used to set TalkBack's service preferences. */
@@ -1015,17 +1013,6 @@ public class TalkBackPreferencesActivity extends Activity {
         return;
       }

-      if (GooglePlayServicesUtil.isGooglePlayServicesAvailable(activity)
-          != ConnectionResult.SUCCESS) {
-        // Not needed, but playing safe since this is hard to test outside of China
-        playStoreButton.setIntent(null);
-        final PreferenceGroup category =
-            (PreferenceGroup) findPreferenceByResId(R.string.pref_category_miscellaneous_key);
-        if (category != null) {
-          category.removePreference(playStoreButton);
-        }
-      }
-
       if (playStoreButton.getIntent() != null
           && activity
                   .getPackageManager()
