diff --git a/buildSrc/src/main/kotlin/Helpers.kt b/buildSrc/src/main/kotlin/Helpers.kt
index 440dbdf..950a3c1 100644
--- a/buildSrc/src/main/kotlin/Helpers.kt
+++ b/buildSrc/src/main/kotlin/Helpers.kt
@@ -335,8 +335,8 @@ fun Project.setupPlugin(projectName: String) {
         }
 
         splits.abi {
-            isEnable = true
-            isUniversalApk = false
+            isEnable = false
+            isUniversalApk = true
 
             if (targetAbi.isNotBlank()) {
                 reset()
@@ -347,6 +347,7 @@ fun Project.setupPlugin(projectName: String) {
         flavorDimensions("vendor")
         productFlavors {
             create("oss")
+            create("fdroid")
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
@@ -453,8 +454,8 @@ fun Project.setupApp() {
         }
 
         splits.abi {
-            isEnable = true
-            isUniversalApk = false
+            isEnable = false
+            isUniversalApk = true
 
             if (targetAbi.isNotBlank()) {
                 reset()
@@ -466,6 +467,7 @@ fun Project.setupApp() {
         productFlavors {
             create("oss")
             create("expert")
+            create("fdroid")
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
