diff --git a/external/openconnect/android/Makefile b/external/openconnect/android/Makefile
--- a/external/openconnect/android/Makefile
+++ b/external/openconnect/android/Makefile
@@ -51,7 +51,7 @@ FETCH=$(TOPDIR)/fetch.sh
 
 CONFIGURE_ARGS := --host=$(TRIPLET) --prefix=$(OC_SYSROOT) \
 		  --disable-shared --enable-static \
-		  CFLAGS="$(EXTRA_CFLAGS)"
+		  --disable-assembly CFLAGS="$(EXTRA_CFLAGS)"
 
 SOURCE_LIST = $(LIBXML2_SRC)/configure $(GMP_SRC)/configure \
 	$(NETTLE_SRC)/configure $(GNUTLS_SRC)/configure \
