diff --git a/src/org/torproject/android/service/TorResourceInstaller.java b/src/org/torproject/android/service/TorResourceInstaller.java
index 648f752..b5f3449 100644
--- a/src/org/torproject/android/service/TorResourceInstaller.java
+++ b/src/org/torproject/android/service/TorResourceInstaller.java
@@ -90,17 +90,6 @@ public class TorResourceInstaller implements TorServiceConstants {
         //only install if ARM (GoLang only supports ARM for now)
         if (Build.CPU_ABI.contains("arm"))
         {
-	        is = context.getResources().openRawResource(R.raw.obfs4proxy);
-	        outFile = new File(installFolder, OBFSCLIENT_ASSET_KEY);
-	        shell.add(new SimpleCommand(COMMAND_RM_FORCE + outFile.getAbsolutePath())).waitForFinish();
-	        streamToFile(is,outFile, false, true);
-	        setExecutable(outFile);
-	        
-	        is = context.getResources().openRawResource(R.raw.meek);
-	        outFile = new File(installFolder, MEEK_ASSET_KEY);
-	        shell.add(new SimpleCommand(COMMAND_RM_FORCE + outFile.getAbsolutePath())).waitForFinish();
-	        streamToFile(is,outFile, false, true);
-	        setExecutable(outFile);
 	        
 	        cpuPath = "armeabi";
         }
