diff --git a/src/com/commonsware/android/arXiv/SingleItemWindow.java b/src/com/commonsware/android/arXiv/SingleItemWindow.java
index 9b00d85..66d5a18 100644
--- a/src/com/commonsware/android/arXiv/SingleItemWindow.java
+++ b/src/com/commonsware/android/arXiv/SingleItemWindow.java
@@ -304,7 +304,7 @@ public class SingleItemWindow extends Activity implements View.OnClickListener {
                         Log.d("arXiv - ","Storage path: "+success);
 
                         if (fare.exists()) {
-                            pdfPath = "storeagePath+"/";
+                            pdfPath = storagePath+"/";
                             vStorage = true;
                         } else {
                             File efare = new File("/mnt/sdcard/arXiv");
@@ -313,7 +313,7 @@ public class SingleItemWindow extends Activity implements View.OnClickListener {
                                 pdfPath = "/mnt/sdcard/arXiv/";
                                 vStorage = true;
                             } else {
-                                File efare = new File("/emmc/arXiv");
+                                efare = new File("/emmc/arXiv");
                                 efare.mkdir();
                                 if (efare.exists()) {
                                     pdfPath = "/emmc/arXiv/";
