diff --git a/SMSdroid/build.gradle b/SMSdroid/build.gradle
index 42c6688..22ffdea 100644
--- a/SMSdroid/build.gradle
+++ b/SMSdroid/build.gradle
@@ -65,7 +65,7 @@ android {
 dependencies {
     compile 'com.android.support:support-v4:23.1.1'
     compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'com.google.android.gms:play-services-ads:8.4.0'
+//    compile 'com.google.android.gms:play-services-ads:8.4.0'
     compile 'de.ub0r.android.lib:lib:1.1.1'
     compile 'de.ub0r.android.logg0r:logg0r:2.0.0'
     compile 'de.ub0r.android.eucookieconsent:library:0.2.0@aar'
diff --git a/SMSdroid/src/main/AndroidManifest.xml b/SMSdroid/src/main/AndroidManifest.xml
index b29dfac..51f42be 100644
--- a/SMSdroid/src/main/AndroidManifest.xml
+++ b/SMSdroid/src/main/AndroidManifest.xml
@@ -204,13 +204,6 @@
                 android:authorities="de.ub0r.android.smsdroid.SearchProvider"
                 android:syncable="false"/>
 
-        <activity
-                android:name="com.google.android.gms.ads.AdActivity"
-                android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize"/>
-
-        <meta-data android:name="com.google.android.gms.version"
-                   android:value="@integer/google_play_services_version" />
-
         <meta-data
                 android:name="android.app.default_searchable"
                 android:value=".ConversationList"/>
diff --git a/SMSdroid/src/main/java/com.android.gms.ads.AdRequest b/SMSdroid/src/main/java/com.android.gms.ads.AdRequest
new file mode 100644
index 0000000..e69de29
diff --git a/SMSdroid/src/main/java/com/google/android/gms/ads/AdListener.java b/SMSdroid/src/main/java/com/google/android/gms/ads/AdListener.java
new file mode 100644
index 0000000..0d1d2b7
--- /dev/null
+++ b/SMSdroid/src/main/java/com/google/android/gms/ads/AdListener.java
@@ -0,0 +1,6 @@
+package com.google.android.gms.ads;
+
+public class AdListener {
+	public void onAdLoaded() {};
+	public void onAdClosed() {};
+}
diff --git a/SMSdroid/src/main/java/com/google/android/gms/ads/AdRequest.java b/SMSdroid/src/main/java/com/google/android/gms/ads/AdRequest.java
new file mode 100644
index 0000000..675a169
--- /dev/null
+++ b/SMSdroid/src/main/java/com/google/android/gms/ads/AdRequest.java
@@ -0,0 +1,7 @@
+package com.google.android.gms.ads;
+
+public class AdRequest {
+	static public class Builder {
+		public AdRequest build() { return null; }
+        }
+}
diff --git a/SMSdroid/src/main/java/com/google/android/gms/ads/AdView.java b/SMSdroid/src/main/java/com/google/android/gms/ads/AdView.java
new file mode 100644
index 0000000..ba8a1e6
--- /dev/null
+++ b/SMSdroid/src/main/java/com/google/android/gms/ads/AdView.java
@@ -0,0 +1,30 @@
+package com.google.android.gms.ads;
+
+import android.view.View;
+import android.content.Context;
+import android.util.AttributeSet;
+
+public class AdView extends View {
+     public AdView(Context c) { super(c); } 
+     public AdView(Context c,AttributeSet a) { super(c,a); } 
+     public AdView(Context c,AttributeSet a,int i1) { super(c,a,i1); } 
+     public AdView(Context c,AttributeSet a,int i1, int i2) { super(c,a,i1,i2); }
+
+
+
+	private String _adSize = "";
+	private String _adUnitId = "";
+
+	public String getAdSize() { return _adSize; }
+	public void setAdSize(String v) { _adSize=v; }
+	public String getAdUnitId() { return _adUnitId;}
+	public void getAdUnitId(String v) { _adUnitId = v;}
+
+	public void loadAd(Object o) {}
+	public void setAdListener(Object o) { }
+ 	public void destroy() { }	
+	public void pause() {}
+	public void resume() {}
+
+
+}
diff --git a/SMSdroid/src/main/java/com/google/android/gms/ads/InterstitialAd.java b/SMSdroid/src/main/java/com/google/android/gms/ads/InterstitialAd.java
new file mode 100644
index 0000000..24f9efe
--- /dev/null
+++ b/SMSdroid/src/main/java/com/google/android/gms/ads/InterstitialAd.java
@@ -0,0 +1,18 @@
+package com.google.android.gms.ads;
+
+public class InterstitialAd {
+
+	public InterstitialAd() {}
+	public InterstitialAd(Object o) {}
+
+	public void loadAd(Object o) {}
+        public void setAdListener(Object o) { }
+	public void show() {}
+
+	public void setAdUnitId(Object o) {}
+	public boolean isLoaded() { return false; }
+
+
+
+}
+
diff --git a/SMSdroid/src/main/res/values/attrs_play_ads_shim.xml b/SMSdroid/src/main/res/values/attrs_play_ads_shim.xml
new file mode 100644
index 0000000..8dc7db7
--- /dev/null
+++ b/SMSdroid/src/main/res/values/attrs_play_ads_shim.xml
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <declare-styleable name="playAdsShim">
+        <attr name="adUnitId" format="string" />
+        <attr name="adSize" format="string" />
+    </declare-styleable>
+
+</resources>
