diff -ur com.roguetemple.hyperroid/AndroidManifest.xml hyperroid/AndroidManifest.xml
--- com.roguetemple.hyperroid/AndroidManifest.xml	2015-03-11 09:59:01.070147132 +0100
+++ hyperroid/AndroidManifest.xml	2014-03-25 12:36:36.000000000 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
       package="com.roguetemple.hyperroid"
-      android:versionCode="440" android:versionName="4.4">
+      android:versionCode="441" android:versionName="4.4a">
 <!--    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />         -->
     <uses-sdk android:minSdkVersion="7" android:targetSdkVersion="7" />
     <application android:label="@string/app_name"
diff -ur com.roguetemple.hyperroid/src/com/roguetemple/hyperroid/HyperRogue.java hyperroid/src/com/roguetemple/hyperroid/HyperRogue.java
--- com.roguetemple.hyperroid/src/com/roguetemple/hyperroid/HyperRogue.java	2015-03-11 09:52:11.282163474 +0100
+++ hyperroid/src/com/roguetemple/hyperroid/HyperRogue.java	2014-03-25 12:36:38.000000000 +0100
@@ -364,11 +364,12 @@
   	      if(curvol < 1) {
     	    curvol += delta;
     	    if(curvol > 1) curvol = 1;
+    	    if(backgroundmusic != null)
     	    backgroundmusic.setVolume(curvol, curvol);
     	    }
     	  }
     	else if(backgroundmusic == null) {
-    	  int id = 0;
+    	  int id = R.raw.crossroads;
     	  if(curland == 3) id = R.raw.desert; 
     	  if(curland == 4) id = R.raw.icyland; 
     	  if(curland == 5) id = R.raw.caves; 
