--- a/BiglyBT/src/coreFlavor/java/com/biglybt/android/core/az/PlatformManagerImpl.java
+++ b/BiglyBT/src/coreFlavor/java/com/biglybt/android/core/az/PlatformManagerImpl.java
@@ -37,8 +37,6 @@
 import android.os.Environment;
 import android.util.Log;
 
-import dalvik.system.DexClassLoader;
-
 public class PlatformManagerImpl
 	implements PlatformManager
 {
@@ -315,29 +313,6 @@
 					Log.i("Core", class_name + "] dex_path=" + dex_path);
 				}
 
-				String tmp = AETemporaryFileHandler.getTempDirectory().getAbsolutePath();
-
-				DexClassLoader dex_Loader = new DexClassLoader(dex_path, tmp, null,
-						VuzeFile.class.getClassLoader());
-
-				return (dex_Loader.loadClass(class_name));
-
-				/*
-				try{
-					Class cla = Class.forName( "dalvik.system.DexClassLoader" );
-					
-					Object dcl = cla.getConstructor(
-						new Class[]{ String.class, String.class, String.class, ClassLoader.class }).newInstance(
-							new Object[]{ dex_path, tmp, null, VuzeFile.class.getClassLoader() });
-					
-					Method method = cla.getMethod( "loadClass", new Class[]{ String.class } );
-					
-					return((Class<?>)method.invoke( dcl,new Object[]{ class_name }));
-					
-				}catch( Throwable e ){
-					e.printStackTrace();
-				}
-				*/
 			}
 
 			return (loader.loadClass(class_name));
