diff --git a/project.properties b/project.properties
index 79a96c4..910deaf 100644
--- a/project.properties
+++ b/project.properties
@@ -13,4 +13,3 @@ target=android-19
 android.library.reference.1=../ActionBar-PullToRefresh
 android.library.reference.2=../drag-sort-listview
 android.library.reference.3=../SlidingMenu
-android.library.reference.4=../google-play-services_lib
diff --git a/src/org/mariotaku/twidere/activity/MapViewerActivity.java b/src/org/mariotaku/twidere/activity/MapViewerActivity.java
index 23e9fc7..d8d5a3f 100644
--- a/src/org/mariotaku/twidere/activity/MapViewerActivity.java
+++ b/src/org/mariotaku/twidere/activity/MapViewerActivity.java
@@ -26,13 +26,9 @@ import android.support.v4.app.FragmentTransaction;
 import android.view.Menu;
 import android.view.MenuItem;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GooglePlayServicesUtil;
-
 import org.mariotaku.twidere.Constants;
 import org.mariotaku.twidere.R;
 import org.mariotaku.twidere.activity.support.TwidereSwipeBackActivity;
-import org.mariotaku.twidere.fragment.support.NativeMapFragment;
 import org.mariotaku.twidere.fragment.support.WebMapFragment;
 import org.mariotaku.twidere.util.MapInterface;
 import org.mariotaku.twidere.util.ThemeUtils;
@@ -92,13 +88,13 @@ public class MapViewerActivity extends TwidereSwipeBackActivity implements Const
 			finish();
 			return;
 		}
-		final Fragment fragment = isNativeMapSupported() ? new NativeMapFragment() : new WebMapFragment();
+		final Fragment fragment = new WebMapFragment();
 		fragment.setArguments(bundle);
 		final FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
 		ft.replace(android.R.id.content, fragment).commit();
 	}
 
 	private boolean isNativeMapSupported() {
-		return GooglePlayServicesUtil.isGooglePlayServicesAvailable(this) == ConnectionResult.SUCCESS;
+		return false;
 	}
 }
