diff --git a/build.gradle.kts b/build.gradle.kts
index 79564c1..c8218fc 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -30,11 +30,7 @@ subprojects {
 
     repositories {
         mavenCentral()
-        maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
         google()
-
-        // for Flatpak offline no-network-access build
-        maven(uri(rootDir.resolve("offline-repository")))
     }
 }
 
@@ -85,18 +81,6 @@ subprojects {
                         url = "https://github.com/archivekeep/archivekeep"
                     }
                 }
-
-                repositories {
-                    maven {
-                        name = "LocalOutput"
-                        url = uri(rootProject.layout.buildDirectory.dir("maven-publish-output"))
-                    }
-                    maven {
-                        name = "CentralSnapshots"
-                        url = uri("https://central.sonatype.com/repository/maven-snapshots/")
-                        credentials(PasswordCredentials::class)
-                    }
-                }
             }
         }
     }
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 8f86c6c..02cd0c2 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -3,8 +3,6 @@ pluginManagement {
         mavenCentral()
         gradlePluginPortal()
         google()
-
-        maven(uri(rootDir.resolve("./offline-repository")))
     }
 }
 
