diff --git a/main/src/main/java/com/google/android/apps/dashclock/configuration/ConfigureExtensionsFragment.java b/main/src/main/java/com/google/android/apps/dashclock/configuration/ConfigureExtensionsFragment.java
index 4016a68..6007dcf 100644
--- a/main/src/main/java/com/google/android/apps/dashclock/configuration/ConfigureExtensionsFragment.java
+++ b/main/src/main/java/com/google/android/apps/dashclock/configuration/ConfigureExtensionsFragment.java
@@ -173,8 +173,7 @@ public class ConfigureExtensionsFragment extends Fragment implements
             @Override
             public boolean onTouch(View view, MotionEvent motionEvent) {
                 return dragSortController.onTouch(view, motionEvent)
-                        || (!dragSortController.isDragging()
-                        && swipeDismissTouchListener.onTouch(view, motionEvent));
+                        || swipeDismissTouchListener.onTouch(view, motionEvent);
 
             }
         });
