Disable the trial version check.

diff --git a/src/org/fox/ttrss/OnlineActivity.java b/src/org/fox/ttrss/OnlineActivity.java
index ef94131..38ecbee 100644
--- a/src/org/fox/ttrss/OnlineActivity.java
+++ b/src/org/fox/ttrss/OnlineActivity.java
@@ -453,8 +453,7 @@ public class OnlineActivity extends CommonActivity {
 	}
 	
 	public void checkTrial(boolean notify) {
-		boolean isTrial = getPackageManager().checkSignatures(
-				getPackageName(), "org.fox.ttrss.key") != PackageManager.SIGNATURE_MATCH;
+		boolean isTrial = false;
 
 		if (isTrial) {
 			long firstStart = m_prefs.getLong("date_firstlaunch_trial", -1);
