diff --git a/uncover/build.gradle b/uncover/build.gradle
index 29f961a..c0a3963 100644
--- a/uncover/build.gradle
+++ b/uncover/build.gradle
@@ -145,14 +145,6 @@ publishing {

     repositories {
         mavenLocal()
-        maven {
-          credentials {
-            username System.env.ANDVIANE_USERNAME
-            password System.env.ANDVIANE_PASSWORD
-          }
-          //url 'https://oss.sonatype.org/content/repositories/snapshots'
-          url 'https://oss.sonatype.org/service/local/staging/deploy/maven2'
-        }
     }
 }
