diff --git a/CacheCleaner/build.gradle b/CacheCleaner/build.gradle
index e629920..a8ac4e4 100644
--- a/CacheCleaner/build.gradle
+++ b/CacheCleaner/build.gradle
@@ -15,6 +15,11 @@ repositories {
 android {
     compileSdkVersion 21
     buildToolsVersion "21.1.2"
+
+    compileOptions {
+           sourceCompatibility JavaVersion.VERSION_1_7
+           targetCompatibility JavaVersion.VERSION_1_7
+    }
 }
 
 dependencies {
