diff -r 431da48bfb10 -r c7c8c7aa3f67 src/org/kost/externalip/ExternalIP.java
--- a/src/org/kost/externalip/ExternalIP.java	Thu Jun 09 01:11:36 2011 +0200
+++ b/src/org/kost/externalip/ExternalIP.java	Fri Oct 26 14:25:52 2012 +0100
@@ -83,10 +83,11 @@
         ip.setText("Please wait...");  
         try {
         	HttpClient httpclient = new DefaultHttpClient();
-        	HttpGet httpget = new HttpGet("http://wiki.iti-lab.org/ip.php");
+        	// HttpGet httpget = new HttpGet("http://wiki.iti-lab.org/ip.php");
         	// HttpGet httpget = new HttpGet("http://whatismyip.everdot.org/ip");
         	// HttpGet httpget = new HttpGet("http://whatismyip.com.au/");
         	// HttpGet httpget = new HttpGet("http://www.whatismyip.org/");
+        	HttpGet httpget = new HttpGet("http://checkip.dyndns.org");
         	HttpResponse response;
         	
         	response = httpclient.execute(httpget);
