--- a/yabause/src/android/android.cmake
+++ b/yabause/src/android/android.cmake
@@ -5,7 +5,10 @@ SET(CMAKE_C_COMPILER   arm-linux-androideabi-gcc)
 SET(CMAKE_CXX_COMPILER arm-linux-androideabi-g++)
 SET(CMAKE_ASM-ATT_COMPILER arm-linux-androideabi-as)
 
-SET(CMAKE_FIND_ROOT_PATH /home/guillaume/projects/android/toolchain/sysroot/usr/)
+SET(CMAKE_FIND_ROOT_PATH $ENV{ANDROID_NDK}/platforms/android-8/arch-arm/usr/)
+
+set(CMAKE_C_FLAGS "--sysroot=$ENV{ANDROID_NDK}/platforms/android-8/arch-arm" CACHE STRING "GCC flags" FORCE)
+set(CMAKE_CXX_FLAGS "--sysroot=$ENV{ANDROID_NDK}/platforms/android-8/arch-arm" CACHE STRING "G++ flags" FORCE)
 
 SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
