diff --git a/android/app/build.gradle b/android/app/build.gradle
index 2ed9d8163..0d690be0c 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -247,14 +247,14 @@ dependencies {
     implementation project(':watermelondb')
     implementation project(":reactnativekeyboardinput")
     implementation project(':@react-native-community_viewpager')
-    playImplementation project(':reactnativenotifications')
-    playImplementation project(':@react-native-firebase_app')
-    playImplementation project(':@react-native-firebase_analytics')
-    playImplementation project(':@react-native-firebase_crashlytics')
+    // playImplementation project(':reactnativenotifications')
+    // playImplementation project(':@react-native-firebase_app')
+    // playImplementation project(':@react-native-firebase_analytics')
+    // playImplementation project(':@react-native-firebase_crashlytics')
     implementation fileTree(dir: "libs", include: ["*.jar"])
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
-    playImplementation "com.google.firebase:firebase-messaging:18.0.0"
+    // playImplementation "com.google.firebase:firebase-messaging:18.0.0"
     implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
     debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
       exclude group:'com.facebook.fbjni'
diff --git a/android/build.gradle b/android/build.gradle
index ae769e0c7..c1dbe393c 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -24,9 +24,9 @@ buildscript {
         mavenLocal()
         google()
         jcenter()
-        maven {
-            url 'https://maven.fabric.io/public'
-        }
+        // maven {
+        //     url 'https://maven.fabric.io/public'
+        // }
     }
 
     dependencies {
diff --git a/android/settings.gradle b/android/settings.gradle
index ed2f78c10..3ddc4b5ed 100644
--- a/android/settings.gradle
+++ b/android/settings.gradle
@@ -4,17 +4,17 @@ includeUnimodulesProjects()
 rootProject.name = 'RocketChatRN'
 include ':watermelondb'
 project(':watermelondb').projectDir = new File(rootProject.projectDir, '../node_modules/@nozbe/watermelondb/native/android')
-include ':reactnativenotifications'
-project(':reactnativenotifications').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-notifications/android/app')
+// include ':reactnativenotifications'
+// project(':reactnativenotifications').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-notifications/android/app')
 include ':reactnativekeyboardinput'
 project(':reactnativekeyboardinput').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-keyboard-input/lib/android')
 include ':@react-native-community_viewpager'
 project(':@react-native-community_viewpager').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-community/viewpager/android')
-include ':@react-native-firebase_app'
-project(':@react-native-firebase_app').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/app/android')
-include ':@react-native-firebase_analytics'
-project(':@react-native-firebase_analytics').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/analytics/android')
-include ':@react-native-firebase_crashlytics'
-project(':@react-native-firebase_crashlytics').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/crashlytics/android')
+// include ':@react-native-firebase_app'
+// project(':@react-native-firebase_app').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/app/android')
+// include ':@react-native-firebase_analytics'
+// project(':@react-native-firebase_analytics').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/analytics/android')
+// include ':@react-native-firebase_crashlytics'
+// project(':@react-native-firebase_crashlytics').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-firebase/crashlytics/android')
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
diff --git a/app/notifications/push/push.android.js b/app/notifications/push/push.android.js
index d83184594..5ba6a2ef7 100644
--- a/app/notifications/push/push.android.js
+++ b/app/notifications/push/push.android.js
@@ -1,4 +1,4 @@
-import { NotificationsAndroid, PendingNotifications } from 'react-native-notifications';
+const { NotificationsAndroid, PendingNotifications } = {};
 
 class PushNotification {
 	constructor() {
@@ -6,11 +6,11 @@ class PushNotification {
 		this.onNotification = null;
 		this.deviceToken = null;
 
-		NotificationsAndroid.setRegistrationTokenUpdateListener((deviceToken) => {
+		NotificationsAndroid?.setRegistrationTokenUpdateListener((deviceToken) => {
 			this.deviceToken = deviceToken;
 		});
 
-		NotificationsAndroid.setNotificationOpenedListener((notification) => {
+		NotificationsAndroid?.setNotificationOpenedListener((notification) => {
 			this.onNotification(notification);
 		});
 	}
@@ -24,8 +24,8 @@ class PushNotification {
 	configure(params) {
 		this.onRegister = params.onRegister;
 		this.onNotification = params.onNotification;
-		NotificationsAndroid.refreshToken();
-		return PendingNotifications.getInitialNotification();
+		NotificationsAndroid?.refreshToken();
+		return PendingNotifications?.getInitialNotification();
 	}
 }
 
diff --git a/app/utils/log/index.js b/app/utils/log/index.js
index da1be54a9..38be41872 100644
--- a/app/utils/log/index.js
+++ b/app/utils/log/index.js
@@ -1,17 +1,9 @@
-import firebaseAnalytics from '@react-native-firebase/analytics';
 import { isFDroidBuild } from '../../constants/environment';
-import config from '../../../config';
 import events from './events';
 
-const analytics = firebaseAnalytics || '';
-let bugsnag = '';
-let crashlytics;
-
-if (!isFDroidBuild) {
-	const { Client } = require('bugsnag-react-native');
-	crashlytics = require('@react-native-firebase/crashlytics').default;
-	bugsnag = new Client(config.BUGSNAG_API_KEY);
-}
+const analytics = () => {};
+const crashlytics = () => {};
+const bugsnag = {};
 
 export { analytics };
 export const loggerConfig = bugsnag.config;
diff --git a/react-native.config.js b/react-native.config.js
index cbd0de1b6..56cbf0ff9 100644
--- a/react-native.config.js
+++ b/react-native.config.js
@@ -30,6 +30,11 @@ module.exports = {
 			platforms: {
 				android: null
 			}
+		},
+		'bugsnag-react-native': {
+			platforms: {
+				android: null
+			}
 		}
 	}
 };
