diff --git a/.classpath b/.classpath
diff --git a/src/pro/dbro/bart/BART.java b/src/pro/dbro/bart/BART.java
index 211e62b..e7aee0e 100644
--- a/src/pro/dbro/bart/BART.java
+++ b/src/pro/dbro/bart/BART.java
@@ -13,7 +13,7 @@ public class BART {
 	public final static String API_ROOT = "http://api.bart.gov/api/";
 	// The public BART API key is: MW9S-E7SL-26DU-VV8V
 	// To obtain your own key, see http://api.bart.gov/api/register.aspx
-	public final static String API_KEY = SECRETS.BART_API_KEY;
+	public final static String API_KEY = "MW9S-E7SL-26DU-VV8V";
 	
 	// *****  BART Route ETA Thresholds  *****
 	// After a day's service has concluded, BART will return scheduled trains for the next day 
diff --git a/src/pro/dbro/bart/BartRouteParser.java b/src/pro/dbro/bart/BartRouteParser.java
index 88b7625..e5ab418 100644
--- a/src/pro/dbro/bart/BartRouteParser.java
+++ b/src/pro/dbro/bart/BartRouteParser.java
@@ -27,7 +27,6 @@ import java.util.Date;
 import java.util.Locale;
 import java.util.TimeZone;
 
-import com.crittercism.app.Crittercism;
 import com.thebuzzmedia.sjxp.XMLParser;
 import com.thebuzzmedia.sjxp.XMLParserException;
 import com.thebuzzmedia.sjxp.rule.DefaultRule;
@@ -166,9 +165,9 @@ public class BartRouteParser extends AsyncTask<String, String, routeResponse> {
 							Log.d("BartRouteParserEndTrip","non-PST coerced parse failed");
 							Log.d("BartRouteParserEndTrip_DateString", "origin: " + originDateStr+" , destination: "+destinationDateStr);
 							Log.d("BartRouteParserEndTripException",e.getClass().toString() + ": " + e.getMessage());
-							Crittercism.leaveBreadcrumb(originDateStr+" , "+destinationDateStr);
-							Crittercism.leaveBreadcrumb(Log.getStackTraceString(e));
-							Crittercism.logHandledException(e);
+							//Crittercism.leaveBreadcrumb(originDateStr+" , "+destinationDateStr);
+							//Crittercism.leaveBreadcrumb(Log.getStackTraceString(e));
+							//Crittercism.logHandledException(e);
 							dateError = true;
 						//}
 					}
