diff --git a/app/build.gradle b/app/build.gradle
new file mode 100644
index 0000000..bba1a96
--- /dev/null
+++ b/app/build.gradle
@@ -0,0 +1,26 @@
+apply plugin: 'com.android.application'
+
+android {
+  compileSdkVersion 19
+  buildToolsVersion '23.0.2'
+  defaultConfig {
+    applicationId "a2dp.Vol"
+    minSdkVersion 15
+    targetSdkVersion 19
+  }
+
+  buildTypes {
+  
+  }
+
+  sourceSets {
+    main {
+        manifest.srcFile 'src/main/AndroidManifest.xml'
+    }
+  }
+
+}
+
+dependencies {
+  compile 'com.android.support:support-v4:21.0.3'
+}
