--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -191,19 +191,6 @@
         //     keyPassword 'android'
         // }
         release {
-            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
-                storeFile file(MYAPP_RELEASE_STORE_FILE)
-                storePassword MYAPP_RELEASE_STORE_PASSWORD
-                keyAlias MYAPP_RELEASE_KEY_ALIAS
-                keyPassword MYAPP_RELEASE_KEY_PASSWORD
-            } else if (System.getenv("BITRISEIO_ANDROID_KEYSTORE_PASSWORD") != null) {
-                storeFile file(System.getenv("HOME") + "/keystores/my_keystore.jks")
-                storePassword System.getenv("BITRISEIO_ANDROID_KEYSTORE_PASSWORD")
-                keyAlias System.getenv("BITRISEIO_ANDROID_KEYSTORE_ALIAS")
-                keyPassword System.getenv("BITRISEIO_ANDROID_KEYSTORE_PRIVATE_KEY_PASSWORD")
-            } else {
-                throw new Exception('Refusing to make release build with wrong keystore')
-            }
         }
     }
 
@@ -226,13 +213,6 @@
             // signingConfig signingConfigs.debug
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
-                signingConfig signingConfigs.release
-            } else if (System.getenv("BITRISEIO_ANDROID_KEYSTORE_PASSWORD") != null) {
-                signingConfig signingConfigs.release
-            } else {
-                throw new Exception('Refusing to make release build with wrong keystore')
-            }
         }
     }
 
