diff --git a/app/build.gradle b/app/build.gradle
index 932a60d..268e3f2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -2,14 +2,12 @@ buildscript {
     repositories {
         jcenter()
         mavenCentral()
-        maven { url 'https://maven.fabric.io/public' }
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:2.0.0'
         classpath 'com.jakewharton.hugo:hugo-plugin:1.2.1'
         classpath 'org.apache.commons:commons-lang3:3.4'
         classpath 'org.ajoberstar:gradle-git:1.1.0'
-        classpath 'io.fabric.tools:gradle:1.15.2'
     }
 }
 
@@ -18,12 +16,11 @@ import com.jbirdvegas.mgerrit.build.utils.VersionHelper
 apply plugin: 'com.android.application'
 apply plugin: 'hugo'
 apply plugin: 'checkstyle'
-apply plugin: 'io.fabric'
 
 repositories {
     jcenter()
     mavenCentral()
-    maven { url 'https://maven.fabric.io/public' }
+    maven { url "https://jitpack.io" }
 }
 
 android {
@@ -41,7 +38,7 @@ android {
         multiDexEnabled true
     }
     dexOptions {
-        javaMaxHeapSize "2g"
+        javaMaxHeapSize "1g"
     }
 
     sourceSets {
@@ -51,10 +48,6 @@ android {
         release {}
     }
     buildTypes {
-        debug {
-            // Disable fabric build ID generation for debug builds
-            ext.enableCrashlytics = false
-        }
         release {
             minifyEnabled false
             signingConfig signingConfigs.release
@@ -92,9 +85,6 @@ android {
 
 dependencies {
     googlePlayCompile files('libs/libGoogleAnalyticsServices.jar')
-    googlePlayCompile('com.crashlytics.sdk.android:crashlytics:2.5.5@aar') {
-        transitive = true
-    }
     compile 'org.jetbrains:annotations:13.0'
 
     compile 'com.android.support:appcompat-v7:23.1.1'
@@ -119,7 +109,7 @@ dependencies {
     compile('com.urswolfer.gerrit.client.rest:gerrit-rest-java-client:0.8.6-android') {
         exclude module: 'httpclient'
     }*/
-    compile files('libs/gerrit-rest-java-client-0.8.8-android.jar')
+    compile 'com.github.p4r4n01d:gerrit-rest-java-client:21e71c1242212e'
     // These are dependencies of the gerrit-rest-java-client library above
     compile 'com.google.guava:guava:18.0'
     compile 'org.apache.httpcomponents:httpclient-android:4.3.5.1'
