diff --git a/madmanLib/build.gradle b/madmanLib/build.gradle
index 4b2e0b4..ac6f045 100644
--- a/madmanLib/build.gradle
+++ b/madmanLib/build.gradle
@@ -1,10 +1,11 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 22
+    compileSdkVersion 23
     buildToolsVersion "23.0.2"
 
     defaultConfig {
+        minSdkVersion 4
         compileOptions {
             sourceCompatibility JavaVersion.VERSION_1_7
             targetCompatibility JavaVersion.VERSION_1_7
diff --git a/qDict/build.gradle b/qDict/build.gradle
index 12fd099..f10584f 100644
--- a/qDict/build.gradle
+++ b/qDict/build.gradle
@@ -26,12 +26,17 @@ android {
     buildTypes {
         release {
             minifyEnabled true
-            proguardFiles 'proguard.cfg'
+            proguardFiles getDefaultProguardFile('proguard-android.txt')
         }
     }
+    
+    lintOptions {
+        abortOnError false
+    }
+
 }
 dependencies {
     compile project(':madmanLib')
     compile project(':appcompat')
-    compile files('libs/nineoldandroids-2.4.0.jar')
+    compile 'com.nineoldandroids:library:2.4.0'
 }
