--- a/openvpn/build.gradle.kts
+++ b/openvpn/build.gradle.kts
@@ -102,9 +102,6 @@
         }
     }

-    signingConfigs {
-        create("release") {}
-    }

     lintOptions {
         enable("BackButton", "EasterEgg", "StopShip", "IconExpectedSize", "GradleDynamicVersion", "NewerVersionAvailable")
@@ -114,7 +111,6 @@

     buildTypes {
         getByName("release") {
-            signingConfig = signingConfigs.getByName("release")
         }
     }

@@ -139,20 +135,7 @@
     }
 }

-// ~/.gradle/gradle.properties
-if (project.hasProperty("keystoreFile") &&
-        project.hasProperty("keystorePassword") &&
-        project.hasProperty("keystoreAliasPassword")) {
-    android.signingConfigs.getByName("release") {
-        storeFile = file(project.properties["keystoreFile"] as String)
-        storePassword = project.properties["keystorePassword"] as String
-        keyPassword = project.properties["keystoreAliasPassword"] as String
-        keyAlias = project.properties["keystoreAlias"] as String
-    }
-} else {
     android.buildTypes.getByName("release").signingConfig = null
-}
-

 /* Hack-o-rama but it works good enough and documentation is surprisingly sparse */

