diff --git a/backends/platform/android/org/scummvm/scummvm/Unpacker.java b/backends/platform/android/org/scummvm/scummvm/Unpacker.java
index da76ceb..a750e2c 100644
--- a/backends/platform/android/org/scummvm/scummvm/Unpacker.java
+++ b/backends/platform/android/org/scummvm/scummvm/Unpacker.java
@@ -314,7 +314,7 @@ public class Unpacker extends Activity {
 			.queryBroadcastReceivers(intent, 0);
 		if (PLUGINS_ENABLED && plugins.isEmpty()) {
 			// No plugins installed
-			AlertDialog.Builder alert = new AlertDialog.Builder(this)
+			/*AlertDialog.Builder alert = new AlertDialog.Builder(this)
 				.setTitle(R.string.no_plugins_title)
 				.setMessage(R.string.no_plugins_found)
 				.setIcon(android.R.drawable.ic_dialog_alert)
@@ -348,7 +348,7 @@ public class Unpacker extends Activity {
 										});
 			}
 
-			alert.show();
+			alert.show();*/
 
 		} else {
 			// Already have at least one plugin installed
