diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a3883498..59f9cf93 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -195,9 +195,9 @@ if( ANDROID )
     configure_file(AndroidManifest.xml.in android/AndroidManifest.xml)
     configure_file(android/build.gradle android/build.gradle COPYONLY)
     configure_file(android/gradle/wrapper/gradle-wrapper.properties android/gradle/wrapper/gradle-wrapper.properties COPYONLY)
-    configure_file(android/gradle/wrapper/gradle-wrapper.jar android/gradle/wrapper/gradle-wrapper.jar COPYONLY)
-    configure_file(android/gradlew android/gradlew COPYONLY)
-    configure_file(android/gradlew.bat android/gradlew.bat COPYONLY)
+#    configure_file(android/gradle/wrapper/gradle-wrapper.jar android/gradle/wrapper/gradle-wrapper.jar COPYONLY)
+#    configure_file(android/gradlew android/gradlew COPYONLY)
+#    configure_file(android/gradlew.bat android/gradlew.bat COPYONLY)
     configure_file(android/res/drawable/splash.xml android/res/drawable/splash.xml COPYONLY)
     configure_file(android/res/values/apptheme.xml android/res/values/apptheme.xml COPYONLY)
     configure_file(android/res/values/apptheme.xml android/res/values/apptheme.xml COPYONLY)
@@ -240,14 +240,14 @@ if( ANDROID )
 
     # The android target depends on our own copy of openssl
     set(ANDROID_EXTRA_LIBS
-        ${openssl_SOURCE_DIR}/arm/libcrypto_1_1.so
-        ${openssl_SOURCE_DIR}/arm/libssl_1_1.so
-        ${openssl_SOURCE_DIR}/arm64/libcrypto_1_1.so
-        ${openssl_SOURCE_DIR}/arm64/libssl_1_1.so
-        ${openssl_SOURCE_DIR}/x86/libcrypto_1_1.so
-        ${openssl_SOURCE_DIR}/x86/libssl_1_1.so
-        ${openssl_SOURCE_DIR}/x86_64/libcrypto_1_1.so
-        ${openssl_SOURCE_DIR}/x86_64/libssl_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/arm/libcrypto_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/arm/libssl_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/arm64/libcrypto_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/arm64/libssl_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/x86/libcrypto_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/x86/libssl_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/x86_64/libcrypto_1_1.so
+        ${CMAKE_SOURCE_DIR}/libs/x86_64/libssl_1_1.so
         CACHE INTERNAL ""
         )
 
