Index: jni/sdl-1.3/src/video/android/SDL_fake_stdout.c
===================================================================
--- a/jni/sdl-1.3/src/video/android/SDL_fake_stdout.c	(revision 115)
+++ b/jni/sdl-1.3/src/video/android/SDL_fake_stdout.c	(working copy)
@@ -34,7 +34,7 @@
 	*/
 	return_value = vsnprintf(buff, sizeof(buff), fmt, ap);
 	va_end(ap);
-	__android_log_print(ANDROID_LOG_INFO, "libSDL", buff);
+	__android_log_print(ANDROID_LOG_INFO, "libSDL", "%s", buff);
 	//delete buff;
 	return return_value;
 }
