diff --git a/QuasselDroid/build.gradle b/QuasselDroid/build.gradle
index 2ba1274..1962d7f 100644
--- a/QuasselDroid/build.gradle
+++ b/QuasselDroid/build.gradle
@@ -14,7 +14,6 @@ repositories {
 
 dependencies {
     compile group: 'com.squareup', name: 'otto', version: '1.3.4'
-    compile files('libs/crittercism_v3_0_7_sdkonly.jar')
     compile project(':ActionBarSherlock')
     compile 'com.android.support:support-v4:19.0.+'
 }
@@ -24,7 +23,7 @@ if(project.hasProperty("Android.signing")
 }
 android {
     compileSdkVersion 19
-    buildToolsVersion "19.0.0"
+    buildToolsVersion "19.1.0"
 
     defaultConfig {
         minSdkVersion 7
