diff --git a/build.gradle b/build.gradle
index cc77667a..48dc31ff 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,33 +5,24 @@ def getPackageName() {
     return "org.linphone"
 }
 
-def firebaseEnable() {
-    File googleFile = new File('google-services.json')
-    return googleFile.exists()
-}
-
 buildscript {
-    File googleFile = new File('google-services.json')
     repositories {
+        google()
         jcenter()
         mavenCentral()
         mavenLocal()
-        google()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.0'
-	if (googleFile.exists()) {
-            classpath 'com.google.gms:google-services:3.1.0'
-	}
     }
 }
 
 allprojects {
     repositories {
+        google()
         jcenter()
         mavenCentral()
         mavenLocal()
-        google()
     }
 }
 
@@ -44,28 +35,12 @@ dependencies {
     androidTestImplementation 'com.jayway.android.robotium:robotium-solo:5.6.3'
     androidTestImplementation 'junit:junit:4.12'
     implementation 'org.apache.commons:commons-compress:1.16.1'
-    if (firebaseEnable()) {
-        implementation 'com.google.firebase:firebase-messaging:15.0.2'
-    }
     implementation 'com.android.support:support-v4:27.0.1'
     implementation project(':liblinphone-sdk')
 }
 
-if (firebaseEnable()) {
-    apply plugin: 'com.google.gms.google-services'
-}
-
 ///// Exclude Files /////
 def excludeFiles = []
-// Exclude firebase file if not enable
-if (!firebaseEnable()) {
-    excludeFiles.add('**/Firebase*')
-    println '[Push Notification] Firebase disabled'
-} else {
-    excludeFiles.add('**/gcm*')
-    println '[Push Notification] Firebase enabled'
-}
-
 excludeFiles.add('src/android/org/linphone/tutorials/*.java')
 
 def excludePackage = []
diff --git a/liblinphone-sdk/build.gradle b/liblinphone-sdk/build.gradle
index 9889fe32..628c5f06 100644
--- a/liblinphone-sdk/build.gradle
+++ b/liblinphone-sdk/build.gradle
@@ -2,10 +2,10 @@
 buildDir = 'bin'
 buildscript {
     repositories {
+        google()
         jcenter()
         mavenCentral()
         mavenLocal()
-        google()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.0'
@@ -14,10 +14,10 @@ buildscript {
 
 allprojects {
     repositories {
+        google()
         jcenter()
         mavenCentral()
         mavenLocal()
-        google()
     }
 }
 
