diff --git a/src/ro/weednet/ContactsSync.java b/src/ro/weednet/ContactsSync.java
index 7c62c2b..90e57a2 100644
--- a/src/ro/weednet/ContactsSync.java
+++ b/src/ro/weednet/ContactsSync.java
@@ -22,8 +22,6 @@
  */
 package ro.weednet;
 
-import com.appbrain.AppBrain;
-
 import ro.weednet.contactssync.Constants;
 import ro.weednet.contactssync.activities.Preferences;
 import ro.weednet.contactssync.client.RawContact;
@@ -83,7 +81,6 @@ public class ContactsSync extends Application {
 		reloadPreferences();
 		mMaxPhotoSize = ContactManager.getPhotoPickSize(this);
 		
-		AppBrain.initApp(this);
 	}
 	
 	public SyncType getSyncType() {
diff --git a/src/ro/weednet/contactssync/activities/Preferences.java b/src/ro/weednet/contactssync/activities/Preferences.java
index 294c64f..2ccc59e 100644
--- a/src/ro/weednet/contactssync/activities/Preferences.java
+++ b/src/ro/weednet/contactssync/activities/Preferences.java
@@ -22,8 +22,6 @@
  */
 package ro.weednet.contactssync.activities;
 
-import com.appbrain.AppBrain;
-
 import ro.weednet.ContactsSync;
 import ro.weednet.contactssync.R;
 import ro.weednet.contactssync.authenticator.AuthenticatorActivity;
@@ -170,10 +168,6 @@ public class Preferences extends Activity {
 	public void onBackPressed() {
 		ContactsSync app = ContactsSync.getInstance();
 		
-		if (!app.getDisableAds()) {
-			AppBrain.getAds().maybeShowInterstitial(this);
-		}
-		
 		finish();
 	}
 	
