diff --git a/src/ro/weednet/contactssync/activities/Preferences.java b/src/ro/weednet/contactssync/activities/Preferences.java
index 08e9a4a..df2fe79 100644
--- a/src/ro/weednet/contactssync/activities/Preferences.java
+++ b/src/ro/weednet/contactssync/activities/Preferences.java
@@ -26,8 +26,6 @@ import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.sql.Date;
 
-import com.ironsource.mobilcore.CallbackResponse;
-import com.ironsource.mobilcore.MobileCore;
 
 import ro.weednet.ContactsSync;
 import ro.weednet.ContactsSync.SyncType;
@@ -104,7 +102,6 @@ public class Preferences extends Activity {
 		ft.replace(R.id.settings, mFragment);
 		ft.commit();
 		
-		MobileCore.init(this, "3QBXU338FKE1M2ZSZEH3WRKIXJ0C5", MobileCore.LOG_TYPE.PRODUCTION, MobileCore.AD_UNITS.OFFERWALL);
 	}
 	
 	@Override
@@ -234,12 +231,6 @@ public class Preferences extends Activity {
 		if (app.getDisableAds()) {
 			super.onBackPressed();
 		} else {
-			MobileCore.showOfferWall(this, new CallbackResponse() {
-				@Override
-				public void onConfirmation(TYPE arg0) {
-					finish();
-				}
-			});
 		}
 	}
 	
