diff --git a/AdAway/build.gradle b/AdAway/build.gradle
index 378ce1c..951466d 100644
--- a/AdAway/build.gradle
+++ b/AdAway/build.gradle
@@ -16,28 +16,6 @@ android {
         targetSdkVersion 17
     }
 
-    /*
-     * To sign release build, create file gradle.properties in .gradle/ with this content:
-     *
-     * signingStoreLocation=/home/key.store
-     * signingStorePassword=xxx
-     * signingKeyAlias=alias
-     * signingKeyPassword=xxx
-     */
-    signingConfigs {
-        myConfig {
-            storeFile file(signingStoreLocation)
-            storePassword signingStorePassword
-            keyAlias signingKeyAlias
-            keyPassword signingKeyPassword
-        }
-    }
-
-    buildTypes {
-        release {
-            signingConfig signingConfigs.myConfig
-        }
-    }
 }
 
 /**
@@ -65,4 +43,4 @@ task renameExecutables(type: Copy) {
 }
 
 
-build.dependsOn renameExecutables
\ No newline at end of file
+build.dependsOn renameExecutables
