diff --git a/build-scripts/tuxguitar-android/apk/build.gradle b/build-scripts/tuxguitar-android/apk/build.gradle
index a54a849..de95da8 100644
--- a/build-scripts/tuxguitar-android/apk/build.gradle
+++ b/build-scripts/tuxguitar-android/apk/build.gradle
@@ -4,7 +4,6 @@ def plugins = [
 	'TuxGuitar-android-gervill-plugin',
 	'TuxGuitar-android-midi-plugin',
 	'TuxGuitar-android-browser-saf-plugin',
-	'TuxGuitar-android-gdrive-plugin',
 	'TuxGuitar-compat',
 	'TuxGuitar-gtp',
 	'TuxGuitar-gpx',
@@ -74,7 +73,6 @@ dependencies {
     implementation project(':tuxguitar-android-gervill')
     implementation project(':tuxguitar-android-midi')
     implementation project(':tuxguitar-android-browser-saf')
-    implementation project(':tuxguitar-android-gdrive')
     implementation project(':tuxguitar-android-gervill-soundbank')
     
     implementation 'com.itextpdf:itextpdf:5.5.10'
diff --git a/build-scripts/tuxguitar-android/apk/src/main/AndroidManifest.xml b/build-scripts/tuxguitar-android/apk/src/main/AndroidManifest.xml
index 5c88c42..a3fd8d8 100644
--- a/build-scripts/tuxguitar-android/apk/src/main/AndroidManifest.xml
+++ b/build-scripts/tuxguitar-android/apk/src/main/AndroidManifest.xml
@@ -13,11 +13,6 @@
 	<uses-permission android:name="android.permission.RECORD_AUDIO" />
 	<uses-permission android:name="android.permission.WRITE_SETTINGS" />
 	
-	<!-- Required by google drive -->
-	<uses-permission android:name="android.permission.MANAGE_ACCOUNTS" />
-	<uses-permission android:name="android.permission.GET_ACCOUNTS" />
-	<uses-permission android:name="android.permission.USE_CREDENTIALS" />
-	
 	<application
 		android:name="org.herac.tuxguitar.android.application.TGApplication"
 		android:allowBackup="true"
@@ -79,9 +74,5 @@
 				<data android:mimeType="audio/midi" />
 			</intent-filter>
 		</activity>
-		
-		<meta-data
-			android:name="com.google.android.gms.version"
-			android:value="@integer/google_play_services_version" />
 	</application>
-</manifest>
\ No newline at end of file
+</manifest>
diff --git a/build-scripts/tuxguitar-android/settings.gradle b/build-scripts/tuxguitar-android/settings.gradle
index 81f5fe9..7a3d450 100644
--- a/build-scripts/tuxguitar-android/settings.gradle
+++ b/build-scripts/tuxguitar-android/settings.gradle
@@ -4,7 +4,6 @@ include ':apk',
 		':tuxguitar-android-resources',
 		':tuxguitar-android-midi',
 		':tuxguitar-android-browser-saf',
-		':tuxguitar-android-gdrive',
 		':tuxguitar-android-gervill-soundbank'
 
 project(':tuxguitar-android').projectDir = new File(settingsDir, '../../TuxGuitar-android')
@@ -12,5 +11,4 @@ project(':tuxguitar-android-resources').projectDir = new File(settingsDir, '../.
 project(':tuxguitar-android-gervill').projectDir = new File(settingsDir, '../../TuxGuitar-android-gervill')
 project(':tuxguitar-android-midi').projectDir = new File(settingsDir, '../../TuxGuitar-android-midi')
 project(':tuxguitar-android-browser-saf').projectDir = new File(settingsDir, '../../TuxGuitar-android-browser-saf')
-project(':tuxguitar-android-gdrive').projectDir = new File(settingsDir, '../../TuxGuitar-android-gdrive')
 project(':tuxguitar-android-gervill-soundbank').projectDir = new File(settingsDir, '../../TuxGuitar-android-gervill-soundbank')
