diff --git a/src/net/robotmedia/acv/comic/Comic.java b/src/net/robotmedia/acv/comic/Comic.java
index f9b9a3d..b0f7872 100644
--- a/src/net/robotmedia/acv/comic/Comic.java
+++ b/src/net/robotmedia/acv/comic/Comic.java
@@ -81,35 +81,8 @@ public abstract class Comic {
 				} catch (Exception e) {
 					Log.w(TAG, "Failed to load zip comic", e);
 				}
-				if (comic == null || comic.isError()) {
-					try {
-						comic = new RarComic(path);
-					} catch (Exception e) {
-						Log.w(TAG, "Failed to load zip comic as rar comic", e);
-					}
-					if (comic != null && !comic.isError()) {
-						Log.i(TAG, "Loaded rar comic with wrong extension");
-						comic.type = Constants.RAR_EXTENSION;
-					}
-				}
-			} else if (Constants.RAR_EXTENSION.equals(type) || Constants.CBR_EXTENSION.equals(type)) {
-				try {
-					comic = new RarComic(path);
-				} catch (Exception e) {
-					Log.w(TAG, "Failed to load rar comic", e);
-				}
-				if (comic == null || comic.isError()) {
-					try {
-						comic = new ZipComic(path);
-					} catch (Exception e) {
-						Log.w(TAG, "Failed to load rar comic as zip comic", e);
-					}
-					if (comic != null && !comic.isError()) {
-						Log.i(TAG, "Loaded rar comic with wrong extension");
-						comic.type = Constants.ZIP_EXTENSION;
-					}
-				}
-			} else if (FileUtils.isImage(type)) {  
+			}
+			 else if (FileUtils.isImage(type)) {  
 				comic = new FileComic(path);
 			} else if (Constants.ACV_EXTENSION.equals(type)) {
 				comic = new ACVComic(path);
