--- a/app/src/main/java/ui/activity/MainActivity.java	2019-05-05 04:27:18.000000000 +0300
+++ b/app/src/main/java/ui/activity/MainActivity.java	2019-05-05 04:29:28.782248394 +0300
@@ -17,12 +17,8 @@
 import android.view.View;
 import android.widget.Toast;
 
-import com.crashlytics.android.Crashlytics;
-import com.crashlytics.android.ndk.CrashlyticsNdk;
 import com.libopenmw.openmw.R;
 
-import io.fabric.sdk.android.Fabric;
-
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileInputStream;
@@ -50,7 +46,6 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        Fabric.with(this, new Crashlytics(), new CrashlyticsNdk());
         PermissionHelper.getWriteExternalStoragePermission(MainActivity.this);
         setContentView(R.layout.main);
         prefs = PreferenceManager.getDefaultSharedPreferences(this);
@@ -79,14 +74,6 @@
             if (openmwCfg.exists()) {
                 BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(openmwCfg)));
                 String line;
-                Crashlytics.log("openmw.cfg");
-                Crashlytics.log("--------------------------------------------------------------------------------");
-                while ((line = reader.readLine()) != null) {
-                    // Don't log fallback lines, they are mostly useless
-                    if (!line.contains("fallback="))
-                        Crashlytics.log(line);
-                }
-                Crashlytics.log("--------------------------------------------------------------------------------");
             }
         } catch (Exception e) {
             // not a big deal if we can't log the contents
@@ -193,7 +180,6 @@
                 });
             } catch (IOException e) {
                 Log.e(TAG, "Failed to write config files.", e);
-                Crashlytics.logException(e);
             }
         });
         th.start();
