diff --git a/compile.sh b/compile.sh
index c53cee6..a95fd4d 100755
--- a/compile.sh
+++ b/compile.sh
@@ -15,6 +15,10 @@ fi
 # try to detect NDK version
 REL=$(grep -o '^r[0-9]*.*' $ANDROID_NDK/RELEASE.TXT 2>/dev/null|cut -b2-)
 case "$REL" in
+    9*)
+        GCCVER=4.7
+        CXXSTL="/"${GCCVER}
+    ;;
     8?*)
         # we don't use 4.4.3 because it doesn't handle threads correctly.
         # TODO : clang?
