diff --git a/jni/Android.mk b/jni/Android.mk
index 9b168aa..157443d 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -24,9 +24,8 @@ LOCAL_SRC_FILES := \
 	share/sync.cpp
 
 LOCAL_C_INCLUDES += $(JNI_H_INCLUDE)
-LOCAL_LDLIBS += -lpthread
 LOCAL_MODULE := libjni_pinyinime
-LOCAL_SHARED_LIBRARIES := libcutils libutils
+LOCAL_SHARED_LIBRARIES := libutils
 LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_SHARED_LIBRARY)
diff --git a/jni/android/com_android_inputmethod_pinyin_PinyinDecoderService.cpp b/jni/android/com_android_inputmethod_pinyin_PinyinDecoderService.cpp
index dcb63a1..3d232ad 100644
--- a/jni/android/com_android_inputmethod_pinyin_PinyinDecoderService.cpp
+++ b/jni/android/com_android_inputmethod_pinyin_PinyinDecoderService.cpp
@@ -15,7 +15,6 @@
  */
 
 #include <assert.h>
-#include <cutils/log.h>
 #include <jni.h>
 #include <string.h>
 #include <sys/types.h>
@@ -388,11 +387,8 @@ static int registerNativeMethods(JNIEnv* env, const char* className,
     }
 
     clazz = env->FindClass("java/io/FileDescriptor");
-    LOG_FATAL_IF(clazz == NULL, "Unable to find Java class java.io.FileDescriptor");
     gFileDescriptorOffsets.mClass = (jclass) env->NewGlobalRef(clazz);
     gFileDescriptorOffsets.mDescriptor = env->GetFieldID(clazz, "descriptor", "I");
-    LOG_FATAL_IF(gFileDescriptorOffsets.mDescriptor == NULL,
-                 "Unable to find descriptor field in java.io.FileDescriptor");
 
     return JNI_TRUE;
 }
diff --git a/jni/share/userdict.cpp b/jni/share/userdict.cpp
index 2843141..0eac1de 100644
--- a/jni/share/userdict.cpp
+++ b/jni/share/userdict.cpp
@@ -20,7 +20,6 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <cutils/log.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/stat.h>
