diff --git a/android/app/build.gradle b/android/app/build.gradle
index 49f4765..12c93c6 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -254,4 +254,3 @@ task copyDownloadableDepsToLibs(type: Copy) {
 }
 
 apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
-apply from: "../../node_modules/@bugsnag/react-native/bugsnag-react-native.gradle"
diff --git a/android/app/src/main/java/com/standardnotes/MainActivity.java b/android/app/src/main/java/com/standardnotes/MainActivity.java
index a3577f8..1cf968d 100644
--- a/android/app/src/main/java/com/standardnotes/MainActivity.java
+++ b/android/app/src/main/java/com/standardnotes/MainActivity.java
@@ -27,7 +27,7 @@ public class MainActivity extends ReactActivity {
             String packageName = this.getContext().getPackageName();
             Bundle props = new Bundle();
             SharedPreferences settings = this.getContext().getSharedPreferences("react-native", Context.MODE_PRIVATE);
-            String bugsnagOptOut = settings.getString("bugsnagoptout", "true");
+            String bugsnagOptOut = "true";
             props.putBoolean("bugsnagOptOut", bugsnagOptOut.equals("true"));
             if (packageName.equals("com.standardnotes.dev")) {
                 props.putString("env", "dev");
diff --git a/android/app/src/main/java/com/standardnotes/MainApplication.java b/android/app/src/main/java/com/standardnotes/MainApplication.java
index 69c4085..bff7fc2 100644
--- a/android/app/src/main/java/com/standardnotes/MainApplication.java
+++ b/android/app/src/main/java/com/standardnotes/MainApplication.java
@@ -5,8 +5,6 @@ import android.app.Activity;
 import android.content.Context;
 import android.webkit.WebView;
 
-import com.bugsnag.android.BreadcrumbType;
-import com.bugsnag.android.Configuration;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
 import com.facebook.react.ReactInstanceManager;
@@ -19,7 +17,6 @@ import java.lang.reflect.InvocationTargetException;
 import java.util.HashSet;
 import java.util.List;
 
-import com.bugsnag.android.Bugsnag;
 import com.facebook.react.modules.network.OkHttpClientProvider;
 
 import android.annotation.SuppressLint;
@@ -74,23 +71,8 @@ public class MainApplication extends Application implements ReactApplication {
 
     rebuildOkHtttp();

-    Configuration config = Configuration.load(this);
-    config.setEnabledBreadcrumbTypes(new HashSet<BreadcrumbType>() {{
-      add(BreadcrumbType.NAVIGATION);
-      add(BreadcrumbType.STATE);
-      add(BreadcrumbType.PROCESS);
-      add(BreadcrumbType.MANUAL);
-      add(BreadcrumbType.USER);
-      add(BreadcrumbType.LOG);
-      add(BreadcrumbType.ERROR);
-    }});
-
     SharedPreferences settings = getApplicationContext().getSharedPreferences("react-native", Context.MODE_PRIVATE);
-    String bugsnagOptOut = settings.getString("bugsnagoptout", "true");
-
-    if (!bugsnagOptOut.equals("true")) {
-      Bugsnag.start(this, config);
-    }
+    String bugsnagOptOut = "true";
 
     SoLoader.init(this, /* native exopackage */ false);
 
diff --git a/index.js b/index.js
index 25f59b8..6a781f2 100644
--- a/index.js
+++ b/index.js
@@ -1,6 +1,5 @@
 import 'react-native-gesture-handler';
 import 'react-native-url-polyfill/auto';
-import Bugsnag from '@bugsnag/react-native';
 import { enableScreens } from 'react-native-screens';
 import { SNLog } from '@standardnotes/snjs';
 
@@ -12,15 +11,12 @@ import { enableAndroidFontFix } from './src/style/android_text_fix';
 enableScreens();
 
 if (__DEV__ === false) {
-  console.log = () => {};
-  console.warn = () => {};
-  console.error = () => {};
-  SNLog.onError = Bugsnag.notify;
-  SNLog.onLog = Bugsnag.leaveBreadcrumb;
-} else {
-  SNLog.onLog = console.log;
-  SNLog.onError = console.error;
+  console.log = () => { };
+  console.warn = () => { };
+  console.error = () => { };
 }
+SNLog.onLog = console.log;
+SNLog.onError = console.error;
 
 enableAndroidFontFix();
 
diff --git a/src/App.tsx b/src/App.tsx
index 8a7492f..f4ae6e4 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -1,4 +1,3 @@
-import Bugsnag from '@bugsnag/react-native';
 import { ActionSheetProvider } from '@expo/react-native-action-sheet';
 import { MobileApplication } from '@Lib/application';
 import { ApplicationGroup } from '@Lib/application_group';
@@ -132,16 +131,6 @@ export const App = (props: { env: 'prod' | 'dev'; bugsnagOptOut: boolean }) => {
     MobileApplication | undefined
   >();
 
-  useEffect(() => {
-    if (!__DEV__ && !props.bugsnagOptOut) {
-      try {
-        Bugsnag.start();
-      } catch {
-        console.error('Bugnsag failed to start');
-      }
-    }
-  }, [props.bugsnagOptOut]);
-
   useEffect(() => {
     const removeAppChangeObserver = applicationGroupRef.current.addApplicationChangeObserver(
       () => {
diff --git a/src/lib/interface.ts b/src/lib/interface.ts
index 7d20717..44f5950 100644
--- a/src/lib/interface.ts
+++ b/src/lib/interface.ts
@@ -288,18 +288,7 @@ export class MobileDeviceInterface extends DeviceInterface {
   }
 
   async getBugsnagOptedOut() {
-    try {
-      /**
-       * Checking the absense of the 'bugsnagoptout' preference.
-       * If the value is absent, then error reporting is opt-in by default.
-       */
-      const bugsnagOptedOut =
-        (await DefaultPreference.get(BUGSNAG_OPT_OUT_KEY)) ?? 'true';
-
-      return bugsnagOptedOut === 'true';
-    } catch {
-      return true;
-    }
+    return true;
   }
 
   setBusgnagOptedOut(optedOut: true | false) {
