diff --git a/android/BOINC/app/build.gradle b/android/BOINC/app/build.gradle
index 9325404a2a..b832587738 100644
--- a/android/BOINC/app/build.gradle
+++ b/android/BOINC/app/build.gradle
@@ -73,8 +73,8 @@ android {
         applicationId "edu.berkeley.boinc"
         minSdkVersion 16
         targetSdkVersion 28
-        versionCode buildVersionCode().toInteger()
-        versionName buildVersionName()
+        versionCode 71616
+        versionName "7.16.16"
 
         // Required when setting minSdkVersion to 20 or lower
         multiDexEnabled true
