diff --git a/android/BOINC/app/build.gradle b/android/BOINC/app/build.gradle
index 9325404a2a..032a702d94 100644
--- a/android/BOINC/app/build.gradle
+++ b/android/BOINC/app/build.gradle
@@ -94,12 +94,6 @@ android {
             testCoverageEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules-debug.txt'
         }
-        xiaomi_release {
-            initWith release
-        }
-        xiaomi_debug {
-            initWith debug
-        }
     }
     compileOptions {
         sourceCompatibility '1.8'
