--- a/build.gradle
+++ b/build.gradle
@@ -7,14 +7,9 @@
         flatDir {
             dirs 'libs'
         }
-        maven {
-            url 'https://maven.fabric.io/public'
-        }
     }
     dependencies {
-        classpath 'com.google.gms:google-services:4.3.3'
         classpath 'com.android.tools.build:gradle:3.6.3'
-        classpath 'io.fabric.tools:gradle:1.31.2'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
@@ -25,9 +20,6 @@
     repositories {
         google()
         jcenter()
-        maven {
-            url 'https://maven.fabric.io/public'
-        }
         maven { url "https://jitpack.io" }
     }
 }
