Index: gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c
===================================================================
--- a/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c	(revision 40)
+++ b/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c	(working copy)
@@ -45,12 +45,6 @@
 #include <linux/if.h>
 #include <linux/route.h>
 
-struct in6_ifreq {
-  struct in6_addr ifr6_addr;
-  __u32           ifr6_prefixlen;
-  int             ifr6_ifindex; 
-};
-
 struct in6_rtmsg {
   struct in6_addr rtmsg_dst;
   struct in6_addr rtmsg_src;
