Index: gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/lib/cli.c
===================================================================
--- a/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/lib/cli.c	(revision 40)
+++ b/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/lib/cli.c	(working copy)
@@ -29,7 +29,7 @@
   // Print the message, if a message was passed.
   if( message )
   {
-    printf(message);
+    printf("%s", message);
   }
 
   // Print the usage.
Index: gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_redirect.c
===================================================================
--- a/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_redirect.c	(revision 40)
+++ b/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_redirect.c	(working copy)
@@ -60,7 +60,7 @@
 	}
 
 	/* Set the broker address */
-	pal_snprintf(new_broker->address, MAX_REDIRECT_ADDRESS_LENGTH, address);
+	pal_snprintf(new_broker->address, MAX_REDIRECT_ADDRESS_LENGTH, "%s", address);
 
 	/* Validate that the address was set correctly */
 	if (strlen(new_broker->address) != strlen(address)) {
@@ -433,7 +433,7 @@
 		if (strlen(line) && (line[strlen(line) - 1] == '\n' || line[strlen(line) - 1] == '\r')) line[strlen(line) - 1] = '\0';
 
 		/* Copy the line to the buffer */
-		sprintf(buffer, line);
+		sprintf(buffer, "%s", line);
 
 		/* We found a server, so stop looking */
 		found_server = 1;
Index: gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c
===================================================================
--- a/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c	(revision 40)
+++ b/gogoc-android/native/gogoc-1_2-RELEASE/gogoc-tsp/src/tsp/tsp_setup.c	(working copy)
@@ -45,12 +45,6 @@
 #include <linux/if.h>
 #include <linux/route.h>
 
-struct in6_ifreq {
-  struct in6_addr ifr6_addr;
-  __u32           ifr6_prefixlen;
-  int             ifr6_ifindex; 
-};
-
 struct in6_rtmsg {
   struct in6_addr rtmsg_dst;
   struct in6_addr rtmsg_src;
