diff --git a/mobile/android/search/java/org/mozilla/search/autocomplete/SearchBar.java b/mobile/android/search/java/org/mozilla/search/autocomplete/SearchBar.java
--- a/mobile/android/search/java/org/mozilla/search/autocomplete/SearchBar.java
+++ b/mobile/android/search/java/org/mozilla/search/autocomplete/SearchBar.java
@@ -48,16 +48,18 @@ public class SearchBar extends FrameLayo
     private boolean active;
 
     public interface TextListener {
         public void onChange(String text);
         public void onSubmit(String text);
         public void onFocusChange(boolean hasFocus);
     }
 
+    // Deprecation warnings suppressed to allow building with API level 22
+    @SuppressWarnings("deprecation")
     public SearchBar(Context context, AttributeSet attrs) {
         super(context, attrs);
 
         LayoutInflater.from(context).inflate(R.layout.search_bar, this);
 
         editText = (EditText) findViewById(R.id.edit_text);
         editText.addTextChangedListener(new TextWatcher() {
             @Override

