diff --git a/wonderdroid/jni/wswan/jni.c b/wonderdroid/jni/wswan/jni.c
index 666a5d1..22524ad 100644
--- a/wonderdroid/jni/wswan/jni.c
+++ b/wonderdroid/jni/wswan/jni.c
@@ -75,7 +75,7 @@ JNIEXPORT void JNICALL Java_uk_org_cardboardbox_wonderdroid_WonderSwan_load(JNIE
 	}
 
 	snprintf(temp, sizeof(temp), "Loading %s", str);
-	LOGD(temp);
+	LOGD("%s", temp);
 	FILE* file = fopen(str, "r");
 	if (file != NULL) {
 		LOGD("The file loaded!!!");
