From 2d91537327042f0555984fdb32c7f2abe1c2e848 Mon Sep 17 00:00:00 2001
From: Marcus Hoffmann <m.hoffmann@cartelsol.com>
Date: Sat, 2 Sep 2017 17:51:20 +0200
Subject: [PATCH] remove fabric.io and crashlytics

---
 MPDroid/build.gradle | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/MPDroid/build.gradle b/MPDroid/build.gradle
index 40d6285c..2b7df2f0 100644
--- a/MPDroid/build.gradle
+++ b/MPDroid/build.gradle
@@ -17,21 +17,9 @@ import java.awt.HeadlessException
  */
 
 apply plugin: 'com.android.application'
-apply plugin: 'io.fabric'
-
-buildscript {
-    repositories {
-        maven { url 'https://maven.fabric.io/public' }
-    }
-
-    dependencies {
-        classpath 'io.fabric.tools:gradle:1.+'
-    }
-}
 
 repositories {
     mavenCentral()
-    maven { url 'https://maven.fabric.io/public' }
 }
 
 def gitShortHash() {
@@ -126,9 +114,6 @@ dependencies {
     compile project(':JMPDCommAndroid')
     // Libraries from maven
     compile 'asia.ivity.android:drag-sort-listview:1.0'
-    closedbitsCompile('com.crashlytics.sdk.android:crashlytics:2.4.0@aar') {
-        transitive = true
-    }
 }
 
 class BrowseJavadoc implements TaskExecutionListener {
-- 
2.11.0

