From 1af84e559618e239d325c51922057a76d6a11211 Mon Sep 17 00:00:00 2001
From: Marcus Hoffmann <m.hoffmann@cartelsol.com>
Date: Sat, 2 Sep 2017 18:00:01 +0200
Subject: [PATCH] remove versionName logic

---
 MPDroid/build.gradle | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/MPDroid/build.gradle b/MPDroid/build.gradle
index 40d6285c..f120f414 100644
--- a/MPDroid/build.gradle
+++ b/MPDroid/build.gradle
@@ -34,7 +34,7 @@ android {
         minSdkVersion 16
         targetSdkVersion 23
         versionCode 58
-        versionName "1.08 Beta 4 " + gitShortHash()
+        versionName "1.08 Beta 4"
     }
 
     lintOptions {
@@ -60,7 +60,7 @@ android {
         }
 
         foss {
-            versionName defaultConfig.versionName + "-f"
+            versionName defaultConfig.versionName
             ext.enableCrashlytics = false
         }
     }
-- 
2.11.0

