diff --git a/build.gradle b/build.gradle
index fb2e26c..81bfabe 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,5 @@
 buildscript {
     repositories {
-        maven { url 'https://dl.google.com/dl/android/maven2/' }
         jcenter()
         google()
     }
@@ -17,10 +16,6 @@ allprojects {
         maven { url 'https://maven.google.com' }
         maven { url "https://jitpack.io" }
         mavenCentral()
-        maven {
-            url 'http://www.idescout.com/maven/repo/'
-            name 'IDEScout, Inc.'
-        }
     }
 }

@@ -30,7 +25,6 @@ subprojects {
     repositories {
         mavenCentral()
         maven { url "https://jitpack.io" }
-        maven { url "http://dl.bintray.com/arturbosch/code-analysis" }
         google()
     }
     configurations {
