diff --git a/lib/bluetooth/devices/value_formatters/ValueFormatter.dart b/lib/bluetooth/devices/value_formatters/ValueFormatter.dart
index f525c55..351e83e 100644
--- a/lib/bluetooth/devices/value_formatters/ValueFormatter.dart
+++ b/lib/bluetooth/devices/value_formatters/ValueFormatter.dart
@@ -2,7 +2,7 @@ import 'package:mighty_plug_manager/bluetooth/devices/value_formatters/Frequency
 import 'package:mighty_plug_manager/bluetooth/devices/value_formatters/SwitchFormatters.dart';
 
 import 'TempoFormatter.dart';
-import 'decibelFormatter.dart';
+import 'DecibelFormatter.dart';
 import 'PercentageFormatter.dart';
 
 enum InputType { SliderInput, SwitchInput }
