From e44ab05c1a32a2dae4d38ce258f8c71ff95fe416 Mon Sep 17 00:00:00 2001
From: Winston Smith <winston.smith@nurfuerspam.de>
Date: Wed, 22 Jan 2014 01:06:49 +0100
Subject: [PATCH] added NDK_APP_DST_DIR

---
 jni/Android.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/jni/Android.mk b/jni/Android.mk
index d48786c..58868be 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -13,6 +13,7 @@
 # limitations under the License.
 #
 LOCAL_PATH := $(call my-dir)
+NDK_APP_DST_DIR := assets
 
 include $(CLEAR_VARS)
 
-- 
1.8.1.5

