diff --git a/jni/Android.mk b/jni/Android.mk
index d48786c..4f1e2f2 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -13,6 +13,7 @@
 # limitations under the License.
 #
 LOCAL_PATH := $(call my-dir)
+NDK_APP_DST_DIR := src/main/assets
 
 include $(CLEAR_VARS)
 
@@ -20,3 +21,12 @@ LOCAL_MODULE    := scan
 LOCAL_SRC_FILES := scan.c
 
 include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE    := scan5
+LOCAL_SRC_FILES := scan.c
+LOCAL_CFLAGS += -fPIE
+LOCAL_LDFLAGS += -fPIE -pie
+
+include $(BUILD_EXECUTABLE)
diff --git a/jni/Application.mk b/jni/Application.mk
index e69de29..7b5fd9c 100644
--- a/jni/Application.mk
+++ b/jni/Application.mk
@@ -0,0 +1,2 @@
+APP_ABI := armeabi
+APP_PIE := false
