diff --git a/Superuser/src/com/koushikdutta/superuser/AboutFragment.java b/Superuser/src/com/koushikdutta/superuser/AboutFragment.java
index 18203b6..1be806c 100644
--- a/Superuser/src/com/koushikdutta/superuser/AboutFragment.java
+++ b/Superuser/src/com/koushikdutta/superuser/AboutFragment.java
@@ -39,80 +39,17 @@ public class AboutFragment extends BetterListFragment {
         catch (NameNotFoundException e) {
         }
         
+        final String uri = "https://f-droid.org/repository/browse/?fdid=org.fdroid.superuser";
         addItem(R.string.about, new ListItem(getInternal(), getString(R.string.superuser), version, R.drawable.ic_launcher) {
             @Override
             public void onClick(View view) {
                 super.onClick(view);
                 Intent i = new Intent();
-                i.setData(Uri.parse("market://details?id=com.koushikdutta.superuser"));
-                startActivity(i);
-            }
-        });
-        
-        addItem(R.string.about, new ListItem(getInternal(), "Koushik Dutta", "@koush", R.drawable.koush) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
-                i.setClassName("com.twitter.android", "com.twitter.android.ProfileActivity");
-                i.putExtra("screen_name", "koush");
-                try {
-                    startActivity(i);
-                }
-                catch (Exception ex) {
-                    ex.printStackTrace();
-                }
-
-            }
-        });
-        
-        final String uri = "https://github.com/koush/Superuser";
-        addItem(R.string.about, new ListItem(getInternal(), "Github", uri, R.drawable.github) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
                 i.setAction(Intent.ACTION_VIEW);
                 i.setData(Uri.parse(uri));
                 startActivity(i);
             }
         });
 
-        addItem(R.string.apps, new ListItem(getInternal(), "ROM Manager", "The ultimate backup, restore, and ROM installation tool", R.drawable.clockwork512) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
-                i.setData(Uri.parse("market://details?id=com.koushikdutta.rommanager"));
-                startActivity(i);
-            }
-        });        
-        addItem(R.string.apps, new ListItem(getInternal(), "Carbon", "Android's missing backup solution", R.drawable.carbon) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
-                i.setData(Uri.parse("market://details?id=com.koushikdutta.backup"));
-                startActivity(i);
-            }
-        });        
-        addItem(R.string.apps, new ListItem(getInternal(), "DeskSMS", "Seamlessly text message from your email, browser, or instant messenger", R.drawable.desksms) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
-                i.setData(Uri.parse("market://details?id=com.koushikdutta.desktopsms"));
-                startActivity(i);
-            }
-        });
-        addItem(R.string.apps, new ListItem(getInternal(), "Tether", "Use your phone's web connection on a laptop or PC", R.drawable.tether) {
-            @Override
-            public void onClick(View view) {
-                super.onClick(view);
-                Intent i = new Intent();
-                i.setData(Uri.parse("market://details?id=com.koushikdutta.tether"));
-                startActivity(i);
-            }
-        });
     }
 }
