Remove extra Google Analytics

diff --git a/ShowMe/ShowMeX/src/com/showmehills/ShowMeHillsActivity.java b/ShowMe/ShowMeX/src/com/showmehills/ShowMeHillsActivity.java
index 95fe190..a9a8125 100644
--- a/ShowMe/ShowMeX/src/com/showmehills/ShowMeHillsActivity.java
+++ b/ShowMe/ShowMeX/src/com/showmehills/ShowMeHillsActivity.java
@@ -48,8 +48,6 @@ import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.impl.client.DefaultHttpClient;
 
-import com.google.analytics.tracking.android.EasyTracker;
-import com.google.analytics.tracking.android.Tracker;
 import com.showmehills.R;
 
 import android.app.Activity;
@@ -231,8 +229,6 @@ public class ShowMeHillsActivity extends Activity implements IShowMeHillsActivit
 		}
 		super.onStop();
 
-	    EasyTracker.getInstance().activityStop(this); // Add this method.
-
 	}
 
 	@Override
@@ -240,8 +236,6 @@ public class ShowMeHillsActivity extends Activity implements IShowMeHillsActivit
 
 		super.onCreate(savedInstanceState);
 
-	    EasyTracker.getInstance().activityStart(this); // Add this method.
-
         requestWindowFeature(Window.FEATURE_NO_TITLE);
         getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
         
