diff --git a/platform/android/jni/Application.mk b/platform/android/jni/Application.mk
index 60909b6..a15f9f2 100644
--- a/platform/android/jni/Application.mk
+++ b/platform/android/jni/Application.mk
@@ -3,13 +3,13 @@
 # Suppose our base version is X:
 
 # Version X: armeabi
-#APP_PLATFORM=android-8
-#APP_ABI := armeabi
+APP_PLATFORM=android-8
+APP_ABI := armeabi
 
 # Version X+1: armeabi-v7a (Much faster due to the availability of hardware
 # FP, but cannot be run in the emulator).
-APP_PLATFORM=android-8
-APP_ABI := armeabi-v7a
+#APP_PLATFORM=android-8
+#APP_ABI := armeabi-v7a
 
 # Version X+2: x86 (Requires android-9, so a change needs to be made in
 # AndroidManifest.xml too)
